big_tests/tests/mim_c2s_SUITE.erl

```diff
@@ -101,6 +102,14 @@ too_big_stanza_rejected(Config) ->
     escalus:assert(is_stream_end, escalus_client:wait_for_stanza(Alice)),
     true = escalus_connection:wait_for_close(Alice, timer:seconds(1)).
 
+message_sent_to_malformed_jid_results_in_error(Config) ->
+    AliceSpec = escalus_fresh:create_fresh_user(Config, alice),
+    {ok, Alice, _Features} = escalus_connection:start(AliceSpec),
+    escalus_client:send(Alice, escalus_stanza:chat_to(<<"@invalid">>, <<"Hi!">>)),
+    Stanza = escalus_client:wait_for_stanza(Alice),
+    escalus_assert:is_error(Stanza, <<"modify">>, <<"jid-malformed">>),
+    true = escalus_connection:wait_for_close(Alice, timer:seconds(1)).
```

Comment:
MongooseIM should not close the connection if there is a **stanza error**. Connection is closed only for **stream errors**.